spec/controllers/pockets_controller_spec.rb

```diff
@@ -186,4 +185,71 @@ def edit_weight_call(pocket_id, weight, token)
       end
     end
   end
+
+  describe 'PUT #add_weight' do
+    let!(:unweighed_pocket) do
+      create(:unweighed_pocket, organization: organization, collection: collection)
+    end
+    let!(:weighed_pocket) do
+      create(:weighed_pocket, organization: organization, collection: collection)
+    end
+    let!(:device) do
+      create(:device, device_id: '1', device_type: 'android', organization: organization)
+    end
+
+    def add_weight_call(pocket_id, weight, token)
+      @request.headers['ApiKey'] = token
+      put :add_weight, params: { id: pocket_id, weight: weight }, as: :json
+    end
+
+    context 'when inputs are valid' do
+      before(:each) { add_weight_call(unweighed_pocket.id, 24.56, device.auth_token) }
+
+      it 'does add the weight' do
+        expect(json_response[:weight]).to eq 24.56
+      end
+      it 'does change the state to Weighed' do
+        expect(json_response[:state]).to eq 'Weighed'
+      end
+      it 'does return ok' do
+        expect(response).to have_http_status(:ok)
+      end
+      it 'does return the pocket as specified in the serializer' do
+        expect(json_response).to eq serializer.new(unweighed_pocket.reload).as_json
+      end
+    end
+
+    context 'when ApiKey is missing' do
```

Comment:
Here it is better to create an unauthenticated user and try to use the service. If you have any question about this, look at some other test.